### ISSUE TITLE
-Wshift-overflow in drivers/net/wireless/realtek/rtw88/main.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#467]
```
drivers/net/wireless/realtek/rtw88/main.c:524:15: warning: signed shift result (0x3FF00000000) requires 43 bits to represent, but 'int' only has 32 bits [-Wshift-overflow]
                        ra_mask &= RA_MASK_VHT_RATES | RA_MASK_OFDM_IN_VHT;
                                   ^~~~~~~~~~~~~~~~~
drivers/net/wireless/realtek/rtw88/main.c:476:6: note: expanded from macro 'RA_MASK_VHT_RATES'
                                 RA_MASK_VHT_RATES_3SS)
                                 ^~~~~~~~~~~~~~~~~~~~~
drivers/net/wireless/realtek/rtw88/main.c:473:41: note: expanded from macro 'RA_MASK_VHT_RATES_3SS'
#define RA_MASK_VHT_RATES_3SS   (0x3ff000 << 20)
                                 ~~~~~~~~ ^  ~~
drivers/net/wireless/realtek/rtw88/main.c:527:15: warning: signed shift result (0xFF0000000) requires 37 bits to represent, but 'int' only has 32 bits [-Wshift-overflow]
                        ra_mask &= RA_MASK_HT_RATES | RA_MASK_OFDM_IN_HT_5G;
                                   ^~~~~~~~~~~~~~~~
drivers/net/wireless/realtek/rtw88/main.c:470:6: note: expanded from macro 'RA_MASK_HT_RATES'
                                 RA_MASK_HT_RATES_3SS)
                                 ^~~~~~~~~~~~~~~~~~~~
drivers/net/wireless/realtek/rtw88/main.c:467:39: note: expanded from macro 'RA_MASK_HT_RATES_3SS'
#define RA_MASK_HT_RATES_3SS    (0xff000 << 16)
                                 ~~~~~~~ ^  ~~
drivers/net/wireless/realtek/rtw88/main.c:535:15: warning: signed shift result (0x3FF00000000) requires 43 bits to represent, but 'int' only has 32 bits [-Wshift-overflow]
                        ra_mask &= RA_MASK_VHT_RATES | RA_MASK_CCK_IN_VHT |
                                   ^~~~~~~~~~~~~~~~~
drivers/net/wireless/realtek/rtw88/main.c:476:6: note: expanded from macro 'RA_MASK_VHT_RATES'
                                 RA_MASK_VHT_RATES_3SS)
                                 ^~~~~~~~~~~~~~~~~~~~~
drivers/net/wireless/realtek/rtw88/main.c:473:41: note: expanded from macro 'RA_MASK_VHT_RATES_3SS'
#define RA_MASK_VHT_RATES_3SS   (0x3ff000 << 20)
                                 ~~~~~~~~ ^  ~~
drivers/net/wireless/realtek/rtw88/main.c:540:15: warning: signed shift result (0xFF0000000) requires 37 bits to represent, but 'int' only has 32 bits [-Wshift-overflow]
                        ra_mask &= RA_MASK_HT_RATES | RA_MASK_CCK_IN_HT |
                                   ^~~~~~~~~~~~~~~~
drivers/net/wireless/realtek/rtw88/main.c:470:6: note: expanded from macro 'RA_MASK_HT_RATES'
                                 RA_MASK_HT_RATES_3SS)
                                 ^~~~~~~~~~~~~~~~~~~~
drivers/net/wireless/realtek/rtw88/main.c:467:39: note: expanded from macro 'RA_MASK_HT_RATES_3SS'
#define RA_MASK_HT_RATES_3SS    (0xff000 << 16)
                                 ~~~~~~~ ^  ~~
4 warnings generated.
```

**Comment by nathanchance** [comment]
This should be the proper fix I think: https://gist.github.com/51de646110177fb0f13233955231751b

**Comment by nathanchance** [comment]
Patch sent: https://lore.kernel.org/lkml/20190502150209.4475-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
Patch accepted: https://git.kernel.org/pub/scm/linux/kernel/git/kvalo/wireless-drivers-next.git/commit/?id=237b47efcdbc019ccd094ad8b87847dc6fc7fda7

**Comment by nathanchance** [comment]
Fixed as of next-20190503: https://git.kernel.org/next/linux-next/c/237b47efcdbc019ccd094ad8b87847dc6fc7fda7
